client/asset/dcr/rpcwallet.go

```diff
@@ -0,0 +1,526 @@
+	err := w.client.Connect(ctx, false)
+	if err != nil {
+		return fmt.Errorf("Decred Wallet connect error: %w", err)
+	}
+
+	// The websocket client is connected now, so if any of the following checks
+	// fails and we return with a non-nil error, we must shutdown the rpc client
+	// or subsequent reconnect attempts will be met with "websocket client has
+	// already connected".
+	var success bool
+	defer func() {
+		if !success {
+			w.client.Shutdown()
+			w.client.WaitForShutdown()
+		}
+	}()
+
+	// Check the required API versions.
+	versions, err := w.client.Version(ctx)
+	if err != nil {
+		return fmt.Errorf("DCR ExchangeWallet version fetch error: %w", err)
+	}
+
+	ver, exists := versions["dcrwalletjsonrpcapi"]
+	if !exists {
+		return fmt.Errorf("dcrwallet.Version response missing 'dcrwalletjsonrpcapi'")
+	}
+	walletSemver := dex.NewSemver(ver.Major, ver.Minor, ver.Patch)
+	if !dex.SemverCompatible(requiredWalletVersion, walletSemver) {
+		return fmt.Errorf("dcrwallet has an incompatible JSON-RPC version: got %s, expected %s",
+			walletSemver, requiredWalletVersion)
+	}
+	ver, exists = versions["dcrdjsonrpcapi"]
+	if !exists {
+		return fmt.Errorf("dcrwallet.Version response missing 'dcrdjsonrpcapi'")
+	}
+	nodeSemver := dex.NewSemver(ver.Major, ver.Minor, ver.Patch)
+	if !dex.SemverCompatible(requiredNodeVersion, nodeSemver) {
+		return fmt.Errorf("dcrd has an incompatible JSON-RPC version: got %s, expected %s",
+			nodeSemver, requiredNodeVersion)
+	}
+
+	curnet, err := w.client.GetCurrentNet(ctx)
+	if err != nil {
+		return fmt.Errorf("getcurrentnet failure: %w", err)
+	}
+
+	// Set the tipAtConnect
+	_, currentTip, err := w.GetBestBlock(ctx)
+	if err != nil {
+		return fmt.Errorf("error getting best block height: %w", translateRPCCancelErr(err))
+	}
+	atomic.StoreInt64(&w.tipAtConnect, currentTip)
+
+	success = true
+	w.log.Infof("Connected to dcrwallet (JSON-RPC API v%s) proxying dcrd (JSON-RPC API v%s) on %v",
+		walletSemver, nodeSemver, curnet)
+	return nil
+}
+
+// Disconnect shuts down access to the wallet by disconnecting the rpc client.
+// Part of the Wallet interface.
+func (w *rpcWallet) Disconnect() {
+	w.client.Shutdown()
+	w.client.WaitForShutdown()
+}
+
+// Disconnected returns true if the rpc client is not connected.
+// Part of the Wallet interface.
+func (w *rpcWallet) Disconnected() bool {
+	return w.node.Disconnected()
+}
+
+// AccountOwnsAddress uses the validateaddress rpc to check if the provided
+// address belongs to the specified account.
+// Part of the Wallet interface.
+func (w *rpcWallet) AccountOwnsAddress(ctx context.Context, account, address string) (bool, error) {
+	a, err := stdaddr.DecodeAddress(address, w.chainParams)
+	if err != nil {
+		return false, err
+	}
+	va, err := w.node.ValidateAddress(ctx, a)
+	if err != nil {
+		// Work around a bug with dcrwallet that prevents validateaddress for
+		// locked accounts.
+		if isAccountNotEncryptedErr(err) {
+			return false, nil
+		}
```

Comment:
I can't seem to make this happen. Is this still an issue?

Reply:
@chappjc can shed more light, I guess.

Reply:
Was fixed in wallet with https://github.com/decred/dcrwallet/pull/2023.

Can remove this case I think.